src/components/picker/PickerItem.js

```diff
@@ -18,57 +20,58 @@ import Text from '../text';
  */
 const PickerItem = props => {
   const {
-    renderItem,
     value,
     label,
-    onPress,
     disabled,
-    isSelected,
     selectedIcon = Assets.icons.check,
     selectedIconColor = Colors.primary,
     testID
   } = props;
+  const context = useContext(PickerContext);
+  const {renderItem} = context;
+  const isSelected = isItemSelected(value, context.value);
+  const itemLabel = getItemLabel(label, value, props.getItemLabel || context.getItemLabel);
+  const accessibilityProps = {
+    accessibilityState: isSelected ? {selected: true} : undefined,
+    accessibilityHint: 'Double click to select this suggestion',
+    ...Modifiers.extractAccessibilityProps(props)
+  };
 
   useEffect(() => {
     if (_.isPlainObject(value)) {
       console.warn('UILib Picker.Item will stop supporting passing object as value & label (e.g {value, label}) in the next major version. Please pass separate label and value props');
     }
   }, [value]);
 
-  // TODO: deprecate the check for object
-  const _onPress = useCallback(() => {
-    // onPress(_.isObject(value) ? value : {value, label});
-    onPress(value);
-  }, [value, onPress]);
-
-  const onSelectedLayout = useCallback((...args) => {
-    _.invoke(props, 'onSelectedLayout', ...args);
-  }, []);
-
-  const getLabel = () => {
-    if (_.isObject(value)) {
-      return _.invoke(props, 'getItemLabel', value) || _.get(value, 'label');
-    }
-    return label;
-  };
-
   const selectedIndicator = useMemo(() => {
     if (isSelected) {
       return <Image source={selectedIcon} tintColor={disabled ? Colors.dark60 : selectedIconColor}/>;
     }
   }, [isSelected, disabled, selectedIcon, selectedIconColor]);
 
+  const _onPress = useCallback(() => {
+    context.onPress(value);
```

Comment:
don't we need to safely invoke 'onPress' here?

Reply:
nope. 
We own the context therefore we know we pass an `onPress` handler.